File: app/controllers/profesor_controller.py
from fastapi import APIRouter, Depends, status
from sqlalchemy.orm import Session
from typing import List
from app.config.database import get_db
from app.dtos.profesor_dto import ProfesorCreateDTO, ProfesorUpdateDTO, ProfesorResponseDTO
from app.repositories.profesor_repository import ProfesorRepository
from app.services.profesor_service import ProfesorService

router = APIRouter(prefix="/profesores", tags=["Profesores"])

def get_profesor_service(db: Session = Depends(get_db)) -> ProfesorService:
    repo = ProfesorRepository(db)
    return ProfesorService(repo)

@router.get("", response_model=List[ProfesorResponseDTO], summary="Obtener todos los profesores")
def obtener_todos_los_profesores(service: ProfesorService = Depends(get_profesor_service)):
    return service.obtener_todos_los_profesores()

@router.get("/{id}", response_model=ProfesorResponseDTO, summary="Obtener profesor por ID")
def obtener_profesor_por_id(id: int, service: ProfesorService = Depends(get_profesor_service)):
    return service.obtener_profesor_por_id(id)

@router.get("/identificacion/{tipo}/{numero}", response_model=ProfesorResponseDTO, summary="Obtener profesor por Tipo y Número de Identificación")
def obtener_profesor_por_tipo_y_numero(tipo: str, numero: str, service: ProfesorService = Depends(get_profesor_service)):
    return service.obtener_profesor_por_tipo_y_numero(tipo, numero)

@router.post("", response_model=ProfesorResponseDTO, status_code=status.HTTP_201_CREATED, summary="Crear profesor")
def crear_profesor(dto: ProfesorCreateDTO, service: ProfesorService = Depends(get_profesor_service)):
    return service.crear_profesor(dto)

@router.put("/{id}", response_model=ProfesorResponseDTO, summary="Actualizar profesor")
def actualizar_profesor(id: int, dto: ProfesorUpdateDTO, service: ProfesorService = Depends(get_profesor_service)):
    return service.actualizar_profesor(id, dto)

File: hecho con java/src/main/java/com/umanizales/notasya/controllers/ProfesorController.java
package com.umanizales.notasya.controllers;

import com.umanizales.notasya.dtos.ProfesorDTO;
import com.umanizales.notasya.services.ProfesorService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/v1/profesores")
@Tag(name = "Profesores", description = "Endpoints para la gestión de profesores")
@CrossOrigin(origins = "*")
public class ProfesorController {

    private final ProfesorService service;

    public ProfesorController(ProfesorService service) {
        this.service = service;
    }

    @GetMapping
    @Operation(summary = "Obtener todos los profesores")
    public ResponseEntity<List<ProfesorDTO.Response>> obtenerTodos() {
        return ResponseEntity.ok(service.obtenerTodosLosProfesores());
    }

    @GetMapping("/{id}")
    @Operation(summary = "Obtener profesor por ID")
    public ResponseEntity<ProfesorDTO.Response> obtenerPorId(@PathVariable Long id) {
        return ResponseEntity.ok(service.obtenerProfesorPorId(id));
    }

    @GetMapping("/identificacion/{tipo}/{numero}")
    @Operation(summary = "Obtener profesor por tipo y número de identificación")
    public ResponseEntity<ProfesorDTO.Response> obtenerPorIdentificacion(@PathVariable String tipo, @PathVariable String numero) {
        return ResponseEntity.ok(service.obtenerProfesorPorTipoYNumero(tipo, numero));
    }

    @PostMapping
    @Operation(summary = "Crear nuevo profesor")
    public ResponseEntity<ProfesorDTO.Response> crear(@Valid @RequestBody ProfesorDTO.Create dto) {
        return ResponseEntity.status(HttpStatus.CREATED).body(service.crearProfesor(dto));
    }

    @PutMapping("/{id}")
    @Operation(summary = "Actualizar profesor existente")
    public ResponseEntity<ProfesorDTO.Response> actualizar(@PathVariable Long id, @Valid @RequestBody ProfesorDTO.Update dto) {
        return ResponseEntity.ok(service.actualizarProfesor(id, dto));
    }
}
